echo-telemetry/src/test/groovy/com/netflix/spinnaker/echo/telemetry/TelemetryEventListenerSpec.groovy

```diff
@@ -367,6 +367,77 @@ class TelemetryEventListenerSpec extends Specification {
     }
   }
 
+  def "test event has deploy method information happy path"(String deploymentType, String deployVersion) {
+      given:
+      def configProps = new TelemetryConfig.TelemetryConfigProps()
+        .setInstanceId(instanceId)
+        .setSpinnakerVersion(spinnakerVersion)
+        .setDeploymentMethod([
+          "type": deploymentType,
+          "version": deployVersion
+        ])
+      def deployEnumType = DeploymentMethod.Type.valueOf(deploymentType.toUpperCase())
+
+      @Subject
+      def listener = new TelemetryEventListener(service, configProps, registry)
+
+      when:
+      listener.processEvent(validEvent)
+
+      then:
+      1 * service.log(_) >> { List args ->
+        String body = args[0]?.toString()
+        assert body != null
+
+        // Note the handy Groovy feature of import aliasing Event->EventProto
+        EventProto.Builder eventBuilder = EventProto.newBuilder()
+        JsonFormat.parser().merge(body, eventBuilder)
+
+        EventProto e = eventBuilder.build()
+        assert e != null
+
+        SpinnakerInstance s = e.spinnakerInstance
+        assert s != null
+        assert s.deploymentMethod != null
+        assert s.deploymentMethod.type == deployEnumType
+        assert s.deploymentMethod.version == deployVersion
+      }
+
+      where:
+      deploymentType << ["none", "other", "minnaker", "halyard", "armory_halyard", "kubernetes_operator"]
+      deployVersion << ["1.2.3", "3.2.1", "1.0.0", "3.3.3", "2.1.9", "99.99.99"]
+  }
+
+  def "test event, without deploy method info"() {
```

Comment:
Aligned with other comments, I'd prefer to flip this. A full "validEvent" proto has _everything_, and there are tests that confirm that everything still works when _something_ is missing or not set.